### PR TITLE
refactor(aws): Updating launch template rollout flag

### PR DESCRIPTION
**Pull request by jeyrschabu**
- made flag to consider app:account:test
- added an exclude list for apps

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy [hunk 1]
```diff
@@ -387,23 +387,40 @@ class AutoScalingWorker {
...
 
+    // Property flag to turn off launch template feature. Caching agent might require bouncing the java process
     if (!dynamicConfigService.isEnabled("aws.features.launch-templates", false)) {
       log.debug("Launch Template feature disabled via configuration.")
       return false
     }
 
-    // application allow list: app1,app2
+    // This is a comma separated list of applications to exclude
+    String excludedApps = dynamicConfigService
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy:402]: Can now exclude apps that we deem not ready for migrating.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy [hunk 2]
```diff
@@ -387,23 +387,40 @@ class AutoScalingWorker {
...
+      .getConfig(String.class, "aws.features.launch-templates.excluded-applications", "")
+    if (application in excludedApps.split(",")) {
+      return false
+    }
+
+    // Application allow list with the following format:
+    // app1:account:region1,region2,app2:account:region1
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy:409]: ex: `gate:mgmt:us-east-1,keel:mgmt:us-west-2,eu-west-1....`

**Review by aravindmd — APPROVED** [review]
LGTM

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy (hunk 2 above)
Comment (caseyhebebrand) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy:409]: Can this accept `app:account` with no region?

Comment (jeyrschabu) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy:409]: it requires a one or more regions. Can enumerate all regions in there to achieve the same result.

**Pull request merged**